web/api/v1/api_test.go

```diff
@@ -1776,6 +1777,81 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, es storage.E
 			},
 			responseLen: 2,
 		},
+		// With a limit for the number of metadata per metric.
+		{
+			endpoint: api.metricMetadata,
+			query:    url.Values{"limit_per_metric": []string{"1"}},
+			metadata: []targetMetadata{
+				{
+					identifier: "test",
+					metadata: []scrape.MetricMetadata{
```

Comment:
Maybe it's just me, but I would be more confident with slightly more unit tests. One I would like to see is a test when you have multiple `targetMetadata` values for increased [code coverage](https://en.wikipedia.org/wiki/Code_coverage).

Reply:
I think that's a valid concern, test added :)